realm/realm-library/src/androidTest/java/io/realm/services/RemoteTestService.java

```diff
@@ -0,0 +1,176 @@
+
+import io.realm.Realm;
+import io.realm.log.RealmLog;
+
+/**
+ * Helper class for multi-processes support testing.
+ */
+public abstract class RemoteTestService extends Service {
+    // There is no easy way to dynamically ensure step IDs have same value for different processes. So, use the stupid
+    // way.
+    private static int BASE_MSG_ID = 0;
+    protected static int BASE_SIMPLE_COMMIT = BASE_MSG_ID;
+    protected static int BASE_A_LOT_COMMITS  = BASE_SIMPLE_COMMIT + 100;
+
+    public static abstract class Step {
+        public final int message;
+
+        protected Step(int base, int id) {
+            this.message = base + id;
+            stepMap.put(this.message, this);
+        }
+
+        protected abstract void run();
+
+        protected RemoteTestService getService() {
+            return RemoteTestService.thiz;
+        }
+
+        // Pass a null to tell main process that everything is OK.
+        // Otherwise, pass a error String which will be used by assertion in main process.
+        private void response(String error) {
+            try {
+                Message msg = Message.obtain(null, message);
+                if (error != null) {
+                    Bundle bundle = new Bundle();
+                    bundle.putString(BUNDLE_KEY_ERROR, error);
+                    msg.setData(bundle);
+                }
+                thiz.client.send(msg);
+            } catch (RemoteException e) {
+                RealmLog.error(e);
+            }
+        }
+    }
+
+    public static final String BUNDLE_KEY_ERROR = "error";
+    @SuppressLint("UseSparseArrays")
+    private static Map<Integer, Step> stepMap = new HashMap<Integer, Step>();
+    public static RemoteTestService thiz;
+    private final Messenger messenger = new Messenger(new IncomingHandler());
+    private Messenger client;
+    private File rootFolder;
+    public Realm realm;
+
+    public RemoteTestService() {
+        if (thiz != null) {
+            throw new RuntimeException("Only one instance is allowed!");
+        }
+        thiz = this;
+    }
+
+    @Override
+    public void onCreate() {
+        super.onCreate();
+        StrictMode.ThreadPolicy policy = new StrictMode.ThreadPolicy.Builder().permitAll().build();
+        StrictMode.setThreadPolicy(policy);
+        try {
+            rootFolder = File.createTempFile(this.getClass().getSimpleName(), "");
+        } catch (IOException e) {
+            RealmLog.error(e);
+        }
+        //noinspection ResultOfMethodCallIgnored
+        rootFolder.delete();
+        //noinspection ResultOfMethodCallIgnored
+        rootFolder.mkdir();
+
+        Realm.init(getApplicationContext());
+    }
+
+    public File getRoot() {
+        return rootFolder;
+    }
+
+    @Override
+    public IBinder onBind(Intent intent) {
+        return messenger.getBinder();
+    }
+
+    @Override
+    public boolean onUnbind(Intent intent) {
+        stopSelf();
+        recursiveDelete(rootFolder);
```

Comment:
I think it doesn't, doc from `createTempFile()`:
> This method provides only part of a temporary-file facility. To arrange for a file created by this method to be deleted automatically, use the deleteOnExit() method.
> 
> The prefix argument must be at least three characters long. It is recommended that the prefix be a short, meaningful string such as "hjb" or "mail". The suffix argument may be null, in which case the suffix ".tmp" will be used. 

Also, the `TemporaryFolder` is doing the same.